app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -13,13 +13,15 @@ class << model_name
     end
 
     has_many :availability_zones,            :foreign_key => :ems_id, :dependent => :destroy
-    has_many :flavors,                       :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_tenants,                 :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_resource_quotas,         :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_volumes,                 :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_volume_snapshots,        :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_object_store_containers, :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_object_store_objects,    :foreign_key => :ems_id, :dependent => :destroy
+    has_many :cloud_databases,               :foreign_key => :ems_id, :dependent => :destroy
+    has_many :database_flavors,              :foreign_key => :ems_id, :dependent => :destroy, :class_name => 'DatabaseFlavor'
+    has_many :flavors, -> { where.not(:type => DatabaseFlavor.descendants) }, :foreign_key => :ems_id, :dependent => :destroy
```

Comment:
Ugh...I don't like this.  We should probably change Flavor to move down a level in the hierarchy and become something like VmFlavor or something.  It might be ok for this PR, but something we should clean up ASAP.